beacon-chain/execution/engine_client.go

```diff
@@ -315,6 +319,43 @@ func (s *Service) ExecutionBlockByHash(ctx context.Context, hash common.Hash, wi
 	return result, handleRPCError(err)
 }
 
+// ExecutionBlockByHashes fetches a batch of execution engine blocks by hash by calling
+// eth_blockByHash via JSON-RPC.
+func (s *Service) ExecutionBlockByHashes(ctx context.Context, hashes []common.Hash, withTxs bool) ([]*pb.ExecutionBlock, error) {
+	ctx, span := trace.StartSpan(ctx, "powchain.engine-api-client.ExecutionBlockByHashes")
+	defer span.End()
+	numOfHashes := len(hashes)
+	elems := make([]gethRPC.BatchElem, 0, numOfHashes)
+	execBlks := make([]*pb.ExecutionBlock, 0, numOfHashes)
+	errs := make([]error, 0, numOfHashes)
+	if numOfHashes == 0 {
+		return execBlks, nil
+	}
+	for _, h := range hashes {
+		blk := &pb.ExecutionBlock{}
+		err := error(nil)
+		newH := h
```

Comment:
Is `newH := h` necessary? We worry about mutation?

Reply:
If you dont create a new hash value, the hash value referenced will change with each iteration in the loop.